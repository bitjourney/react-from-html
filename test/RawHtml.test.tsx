import "tsconfig-paths/register";
import React from "react";
import renderer from "react-test-renderer";

import { RawHtml, DOMParserContext } from "./entrypoint";

describe("<RawHtml/>", () => {
  it("renders single block of HTML elements correctly", () => {
    const tree = renderer
      .create(
        <DOMParserContext.Provider value={new DOMParser()}>
          <RawHtml html="<p class='hello'>Hello, <strong>React</strong> world!</p>" />
        </DOMParserContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
