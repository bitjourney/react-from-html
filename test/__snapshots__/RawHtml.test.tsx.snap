// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RawHtml/> renders single block of HTML elements correctly 1`] = `
<p
  className="hello"
>
  Hello, 
  <strong>
    React
  </strong>
   world!
</p>
`;
